vcap_utils/vcap_utils/backends/base_openvino.py

```diff
@@ -109,15 +95,117 @@ def prepare_inputs(self, frame: np.ndarray, frame_input_name: str = None) \
+    def parse_detection_results(
+            self, results: np.ndarray,
+            resize: Resize,
+            label_map: Dict[int, str],
+            min_confidence: float = 0.0) -> List[DetectionNode]:
+        """A helper method to take results from a detection-type network.
+        :param results: The inference results from the network
+        :param resize: A Resize object that was used to resize the image to
+        fit into the network originally.
+        :param label_map: A dictionary mapping integers to class_names.
+        :param min_confidence: Filter out detections that have a confidence
+        less than this number.
+        :returns: A list of DetectionNodes, in this case representing bounding
+        boxes.
+        """
+        output_blob_name = self.output_blob_names[0]
+        inference_results = results[output_blob_name]
+
+        _, _, h, w = self.net.inputs[self.input_blob_names[0]].shape
+
+        nodes: List[DetectionNode] = []
+        for result in inference_results[0][0]:
+            # If the first index == 0, that's the end of real predictions
+            # The network always outputs an array of length 200 even if it does
+            # not have that many predictions
+            if result[0] != 0:
+                break
+
+            confidence = float(result[2])
+            if confidence <= min_confidence:
+                continue
+
+            x_min, y_min, x_max, y_max = result[3:7]
+            # x and y in res are in terms of percent of image width/height
+            x_min, x_max = x_min * w, x_max * w
+            y_min, y_max = y_min * h, y_max * h
+            coords = [[x_min, y_min], [x_max, y_min],
+                      [x_max, y_max], [x_min, y_max]]
+
+            class_id = round(result[1])
+            res = DetectionNode(
+                name=label_map[class_id],
+                coords=coords,
+                extra_data={"detection_confidence": confidence})
+            nodes.append(res)
+
+        # Convert the coordinate space of the detections from the
+        # resized frame to the
+        resize.scale_and_offset_detection_nodes(nodes)
+        return nodes
+
+    def batch_predict(self, inputs: List[OV_INPUT_TYPE]) \
+            -> List[object]:
+        """Use the network for inference.
+
+        This function will receive a list of inputs and process them as
+        efficiently as possible, optimizing for throughput.
+        :param inputs: A list of openvino style inputs {input_name: ndarray}
+        :returns: A generator of the networks outputs, yielding them in the
+        same order as the inputs
+        """
 
-            yield results
+        inputs: Deque[Tuple[int, OV_INPUT_TYPE]] = deque(enumerate(inputs))
+        """A queue containing tuples of (frame_id, input) for inference"""
+        requests_in_progress: Dict[int, int] = {}
+        """A dictionary of {request_id: frame_id} for ongoing requests"""
+        unsent_results: Dict[int: Dict] = {}
```

Comment:
Might want to declare what the nested Dict contains

Reply:
If the nested Dict is untyped, it's preferred to use the object (`dict`) to annotate instead of  the type (`Dict`)